### PR TITLE
ensure only published allocations are fetched

### PR DESCRIPTION
**Pull request by thelostone-mc**

### CODE REVIEW

File: src/server/api/routers/voters.ts [hunk 1]
```diff
@@ -47,6 +47,7 @@ export const votersRouter = createTRPCRouter({
             .findMany({
               where: {
                 roundId: ctx.round.id,
+                ballot: { publishedAt: { not: null } },
```
Comment (carlbarrdahl) [inline src/server/api/routers/voters.ts:50]: Will this affect showing the voters list in the admin ui?

File: src/server/api/routers/results.ts [hunk 2]
```diff
@@ -254,7 +254,12 @@ async function generateImpactPayouts(round: Round, db: PrismaClient) {
 
   // Fetch the allocations for the specified round
   const allocations = await db.allocation.findMany({
-    where: { roundId: round.id },
+    where: {
+      roundId: round.id,
+      ballot: {
+        publishedAt: { not: null },
+      }
+    },
```
Comment (carlbarrdahl) [inline src/server/api/routers/results.ts:262]: Nice catch!

**Review by carlbarrdahl — COMMENTED** [review]
Looks good! Just a question if list voters in the admin ui changes.

File: src/server/api/routers/voters.ts (hunk 1 above)
Comment (thelostone-mc) [reply src/server/api/routers/voters.ts:50]: It won't ! Only affects the check-mark icon

Comment (thelostone-mc) [reply src/server/api/routers/voters.ts:50]: @carlbarrdahl Here is an example
image

Comment (carlbarrdahl) [reply src/server/api/routers/voters.ts:50]: Amazing, great work!

**Review by carlbarrdahl — APPROVED** [review]

**Pull request merged**